packages/connect/src/data/firmwareInfo.ts

```diff
@@ -18,9 +29,25 @@ const releases = Object.values(DeviceModelInternal).reduce(
     {} as Record<keyof typeof DeviceModelInternal, FirmwareRelease[] | undefined>,
 );
 
-export const parseFirmwareReleases = (json: any, deviceModel: DeviceModelInternal) => {
-    Object.keys(json).forEach(key => {
-        const release = json[key];
+let firmwareReleasesConfig: FirmwareReleaseConfig['releases'] | undefined;
+let firmwareIntermediaryReleasesConfig:
+    | Record<keyof typeof DeviceModelInternal, IntermediaryReleaseConfig[]>
+    | undefined;
+const bundledReleases: Record<keyof typeof DeviceModelInternal, FirmwareRelease> = {} as Record<
+    keyof typeof DeviceModelInternal,
+    FirmwareRelease
+>;
+
+export const parseFirmwareReleases = (
+    modelReleases: FirmwareRelease[],
+    deviceModel: DeviceModelInternal,
+) => {
+    if (!Array.isArray(modelReleases)) {
+        return;
+    }
+    const [latestRelease] = modelReleases;
```

Comment:
[latestRelease] would be undefined for Model.UNKNOWN. It probably does not matter?